pages/api/eval.ts

```diff
@@ -65,6 +84,26 @@ export async function rankEvalRequest(
         },
       }
     })
+
+  if (name === 'negative') {
+    // For our negative examples, we want to check that the target IDs _don't_
+    // appear in the list of results. Rank_eval doesn't have a neat way of doing
+    // this out of the box, but we know that the response's metric_score will be
+    // 0 if and only if the ID isn't included in the results. If the ID is
+    // included anywhere in the list, the score for the recall metric will be > 0.
+    //
+    // We therefore intercept the result here. If the score is 0, we manually
+    // set it to 1 (ie pass). Otherwise, it's set to 0 (ie fail).
+    Object.values(response.details).forEach((search) => {
+      if (search.metric_score === 0) {
+        search.metric_score = 1
+      } else {
+        search.metric_score = 0
+      }
+    })
+  }
```

Comment:
The reason I like the invert metric bit being somewhere else is you wouldn't have to mutate this score, but rather on the edge of the application i.e. where we calculate the `pass`, we could say

```ts
function passes(rating) {
  return rating.invertScore ? rating.metric_score === 0 : rating.metric_score === 1
}
const pass = ratings.event(passes)
```

Reply:
Also, because we have a kind of hierarchy of passes/fails, it seemed neater to directly augment the lowest level score to reflect our intentions, rather than augmenting every level of the chain for display.